### ISSUE TITLE
Selector for form[formName].registeredFields

### ISSUE THREAD

**Issue by olslash** [erikras/redux-form#2616]
I have a use case where I'd like to be able to get a list of all the fields that have currently been registered on a form. 

Basically I'm trying to do a kind of fancy wizard flow that necessitates having a list of all the field names, and I'd like to avoid typing them out on both the `<Field>`s and in the container where the business logic is happening (keeping them in sync would be problematic as the form grows)

I don't see a selector for ` form[formName].registeredFields` -- is that intentional? If I make my own, is the field name likely to change?

Thanks!

**Comment by gustavohenke** [comment]
> I don't see a selector for form[formName].registeredFields

This is because one doesn't exist.

I marked it as an enhancement, but please follow the template the next time you open an issue here.
It was made for a reason.

**Comment by rizedr** [comment]
This would be very useful when doing submit validation as you won't need to hard code all the field names again. 

The validate function currently receives only `(values, props)` and in props there is no object with all registered fields, thus forcing you to hardcode all fields so you can check if they were filled or not.

Correct me if I am wrong @gustavohenke

**Comment by gustavohenke** [comment]
No, you're right @rizedr. There are no props for this either.

Thinking about this a little better... I recently developed a custom forms feature. The list of fields include whether the field is required, whether it requires emails/dates/etc. So I base my rules on these instead of something else.
I highly recommend you guys to do something like this. It's probably a bad idea to create dynamic rules based on fixed fields.

**Comment by olslash** [comment]
That's what I ended up doing as well, if i understand you -- my wizard is more or less generated from an object defined in the container like

```
  pageOne: [
    fields: { a: { type: 'text', ... }, ... },
    requiredFields: ['a', 'b', 'c']
  ],
  pageTwo: [ ... ]
```

ended up working pretty well and overall I think it made the code more explicit.

**Comment by rizedr** [comment]
@gustavohenke I disagree with you. It would be a good feature especially if you have different submit actions that require different validation, check the example I wrote in #2511. 

It also falls back when you are trying to achieve validations on fields that are defined at run time. For example I'm building a configurator that generates different inputs depending on past entered values.

@olslash For the use case I am working right, I ended up doing almost the same thing, except my constant has 67 fields in it 😄.
